kolibri/plugins/setup_wizard/assets/src/views/SetupWizardIndex.vue

```diff
@@ -100,12 +100,13 @@
 
       this.service.onTransition(state => {
         const stateID = Object.keys(state.meta)[0];
-        let newRoute = state.meta[stateID].route;
-
-        if (newRoute != this.$router.currentRoute.name) {
-          if ('path' in state.meta[stateID])
-            this.$router.push({ name: newRoute, path: state.meta[stateID].path });
-          else this.$router.push(newRoute);
+        if (state.meta[stateID] !== undefined) {
+          let newRoute = state.meta[stateID].route;
+          if (newRoute != this.$router.currentRoute.name) {
+            if ('path' in state.meta[stateID])
+              this.$router.push({ name: newRoute, path: state.meta[stateID].path });
+            else this.$router.push(newRoute);
+          }
```

Comment:
I think that this logic can be resolved by just passing `synchronizeRouteAndMachine` to the `onTransition` method.

I had written the function intending to do this. There is a change that I made to the `meta` properties on the route state configurations where the `meta.route` key maps to a proper _VueRouter route_ flavored object that is shaped like `{ name?: String, path?: String, ... }`. 

Because of this change to the shape of the `meta.route` property, there will never be a `meta[stateId].path` value because `meta[stateId].route` is always one of those objects with a name and/or path.